### PR TITLE
CHORE: updated compass-icon package

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
- updated compass-icon package
- moved @types packages to devDependencies
- moved prettier to devDependencies

This should fix the overapping icon from this issue: https://github.com/mattermost/mattermost-server/issues/19252#issuecomment-1034605734 and is a replacement PR for #9786 

#### Ticket Link
n/a

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
updated @mattermost/compass-icons package to `^0.1.21`
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,11 +7,9 @@
   "private": true,
   "dependencies": {
     "@mattermost/compass-components": "^0.2.12",
-    "@mattermost/compass-icons": "^0.1.10",
+    "@mattermost/compass-icons": "^0.1.21",
```
Comment (M-ZubairAhmed) [inline package.json:10]: Have we checked this patch update does not cause any ui related changes :P

**Review by M-ZubairAhmed — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (michelengelen) [reply package.json:10]: I dont think so ... most of the time there are only icons added and not changed ... Although with v0.1.18 I added SVG components which make it easier to align the icons when not inline

Comment (M-ZubairAhmed) [reply package.json:10]: ok sounds good

**Comment by M-ZubairAhmed** [comment]
@michelengelen may be this should have release note as compass icon update ?

**Review by hmhealey — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @michelengelen 
Tested, looks good to merge. 
- Verified various icons in the app, overlap on `edited` is fixed. 
- [E2E report ](https://automation-dashboard.vercel.app/cycles/16fe733d-47d1-495f-b842-f5c27ffb295e)looks good, no PR related failures.

**Pull request merged**